ldap/servers/plugins/uiduniq/uid.c

```diff
@@ -1311,26 +1311,26 @@ preop_modrdn(Slapi_PBlock *pb)
     }
 
     /* Get the DN of the entry being renamed */
-    err = slapi_pblock_get(pb, SLAPI_MODRDN_TARGET_SDN, &sdn);
+    err = slapi_pblock_get(pb, SLAPI_MODRDN_TARGET_SDN, &sourceSDN);
     if (err) {
         result = uid_op_error(31);
         break;
     }
 
     /* Get superior value - unimplemented in 3.0/4.0/5.0 DS */
-    err = slapi_pblock_get(pb, SLAPI_MODRDN_NEWSUPERIOR_SDN, &superior);
+    err = slapi_pblock_get(pb, SLAPI_MODRDN_NEWSUPERIOR_SDN, &superiorSDN);
     if (err) {
         result = uid_op_error(32);
         break;
     }
 
     /*
      * No superior means the entry is just renamed at
-     * its current level in the tree.  Use the target DN for
+     * its current level in the tree.  Use the source SDN for
      * determining which managed tree this belongs to
      */
-    if (!superior)
-        superior = sdn;
+    if (!superiorSDN)
+        superiorSDN = sourceSDN;
```

Comment:
superiorSDN being SLAPI_MODRDN_NEWSUPERIOR_SDN, if it is NULL shouldn't it be initialized with slapi_sdn_get_parent(sourceSDN, superiorSDN) ?

Reply:
Where later ?
Do you mean when it is used in slapi_entry_rename(e, rdn, deloldrdn, destinationSDN) ?

Reply:
For me it is the last thing I do not understand. Unless I am mistaken, the forth param of slapi_entry_rename should be the parent of the destination entry and not the destination entry itself.